### PR TITLE
[Merged by Bors] - feat: naturality of the `gelfandStarTransform` and `WeakDual.CharacterSpace.homeoEval`

### PR DESCRIPTION
**Pull request by j-loreaux**
---

Because you asked, Johan :smiley: 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/NormedSpace/Star/GelfandDuality.lean [hunk 1]
```diff
@@ -248,3 +248,34 @@ end CharacterSpace
...
+
+φ                  (F ∘ G) φ
+
+|                     |
+V                     V
+
+B  --- η B ---> C(characterSpace ℂ B, ℂ)
+```
+-/
+theorem gelfand_duality {A B : Type*} [NormedCommRing A] [NormedAlgebra ℂ A] [CompleteSpace A]
```
Comment (jcommelin) [inline Mathlib/Analysis/NormedSpace/Star/GelfandDuality.lean:276]:
```suggestion
theorem gelfandStarTransform_naturality {A B : Type*} [NormedCommRing A] [NormedAlgebra ℂ A] [CompleteSpace A]
```

**Review by jcommelin — APPROVED** [review]
LGTM, but a name suggestion

bors d+

File: Mathlib/Analysis/NormedSpace/Star/GelfandDuality.lean [hunk 2]
```diff
@@ -22,12 +23,21 @@ the Gelfand transform is actually spectrum-preserving (`spectrum.gelfandTransfor
...
 and even an equivalence between C⋆-algebras.
 
+Consider the contravariant functors between compact Hausdorff spaces and commutative unital
+C⋆algebras `F : Cpct → CommCStarAlg := X ↦ C(X, ℂ)` and
+`G : CommCStarAlg → Cpct := A → characterSpace ℂ A` whose actions on morphisms are given by
+`WeakDual.CharacterSpace.compContinuousMap` and `ContinuousMap.compStarAlgHom'`, respectively.
+
+Then `η₁ : id → F ∘ G := gelfandStarTransform` and
+`η₂ : id → G ∘ F := WeakDual.CharacterSpace.homeoEval` are the natural isomorphisms implementing
+**Gelfand Duality**, i.e., the (contravariant) equivalnce of these categories.
```
Comment (jcommelin) [inline Mathlib/Analysis/NormedSpace/Star/GelfandDuality.lean:33]:
```suggestion
**Gelfand Duality**, i.e., the (contravariant) equivalence of these categories.
```

File: Mathlib/Analysis/NormedSpace/Star/GelfandDuality.lean [hunk 3]
```diff
@@ -248,3 +257,64 @@ end CharacterSpace
...
+
+f                  (G ∘ F) f
+
+|                     |
+V                     V
+
+Y  --- η Y ---> characterSpace ℂ C(Y, ℂ)
+```
+-/
+def WeakDual.CharacterSpace.homeoEval_naturality {X Y 𝕜 : Type*} [IsROrC 𝕜] [TopologicalSpace X]
```
Comment (jcommelin) [inline Mathlib/Analysis/NormedSpace/Star/GelfandDuality.lean:316]: This shouldn't be a `def`, right?

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**